src/index.cpp

```diff
@@ -1192,9 +1192,19 @@ void Index<T, TagT, LabelT>::search_for_point_and_prune(int location, uint32_t L
     }
 
     prune_neighbors(location, pool, pruned_list, scratch);
-
-    assert(!pruned_list.empty());
-    assert(_final_graph.size() == _max_points + _num_frozen_pts);
+    // skip full dupl vectors
+    NeighborPriorityQueue &best_L_nodes = scratch->best_l_nodes();
+    _full_duplicated =
```

Comment:
_full_duplicated seems to be a common shared variable amongst threads. 
Here you are modifying it concurrently in each thread. would that not lead to race?